src/chart/Treemap.js

```diff
@@ -398,6 +399,33 @@ class Treemap extends PureComponent {
     }
   }
 
+  handleDoubleClick(node) {
+    const { onDoubleClick, type } = this.props;
+    if (type === 'nest' && node.children) {
+      const { width, height, dataKey, aspectRatio } = this.props;
+      const root = computeNode({
+        depth: 0,
+        node: { ...node, x: 0, y: 0, width, height },
+        index: 0,
+        valueKey: dataKey,
+      });
+
+      const formatRoot = squarify(root, aspectRatio);
+
+      const { nestIndex } = this.state;
+      nestIndex.push(node);
+
+      this.setState({
+        formatRoot,
+        currentRoot: root,
+        nestIndex,
+      });
+    }
```

Comment:
I don't think we want to be doing any of this, because the `click` handler will already fire on the first click of a double-click, and that will do whatever this is doing (which I assume is zooming in or something). If anything, maybe we want to suppress a further zoom on the second click, but it's hard to know without an example. So:

1. Does this do something sensible? It's hard to know without a demo or test.
2. Is there even something sensible it could be doing in the context of a treemap, given the issues of a click always firing before a double-click?
3. Given that we're not using the treemap and it's meaningfully different from the other chart types, is this out of scope for what you're trying to achieve? If you add a simple demo and it doesn't act as you'd expect straight away, I'd argue so.

Reply:
Yeah had a play around with the treemap and it does appear to be a much more complex case than the other charts. There would have to be some way of delaying the click just enough so that the double click could be detected without causing too much lag. Would be a load of work and I would agree that it's out of the scope of this change.

Going to back it out.